db/attachment.go

```diff
@@ -37,8 +37,15 @@ var (
 	ErrAttachmentMeta = base.HTTPErrorf(http.StatusBadRequest, "Invalid _attachments")
 )
 
-// AttachmentData holds the attachment key and value bytes.
-type AttachmentData map[string][]byte
+// updatedAttachment represents an attachment that has been modified by a create or update operation.
+type updatedAttachment struct {
+	body    []byte // The attachment data in bytes
+	updated bool   // If this attachment was an update of an existing attachments
+	created bool   // If this attachment was new on the body
+}
+
+// updatedAttachments holds the attachment key and contents of the attachment.
```

Comment:
What is 'the attachment key' here?  If we know whether this is attachment name, attachment digest, attachment document key, it would be helpful to mention that here.